### PR TITLE
[Backport stable/8.1] Fix: Zeebe gateway fails to stream out activated jobs by not respecting the max message size #15784

### PR DESCRIPTION
**Pull request by berkaycanbc**
## Description

Backport of https://github.com/camunda/zeebe/pull/15784 to stable/8.1.

Related issue https://github.com/camunda/zeebe/issues/12778

## Related issues

closes #

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by berkaycanbc** [comment]
@korthout  Please take a close look to the additional commits. 👀

**Comment by berkaycanbc** [comment]
I will check the failures tomorrow. 👍

File: gateway/src/main/java/io/camunda/zeebe/gateway/impl/job/RoundRobinActivateJobsHandler.java [hunk 1]
```diff
@@ -148,11 +151,22 @@
...
           final var response = brokerResponse.getResponse();
-          final ActivateJobsResponse grpcResponse =
+          final JobActivationResult jobActivationResult =
               ResponseMapper.toActivateJobsResponse(brokerResponse.getKey(), response);
+
+          final List<ActivatedJob> jobsToDefer = jobActivationResult.jobsToDefer();
+          if (!jobsToDefer.isEmpty()) {
+            final var jobKeys = jobsToDefer.stream().map(ActivatedJob::getKey).toList();
+            final var jobType = request.getType();
+            final var reason = String.format(MAX_MESSAGE_SIZE_EXCEEDED_MSG);
```
[the remark this answers is not in the document]
Comment (berkaycanbc) [reply gateway/src/main/java/io/camunda/zeebe/gateway/impl/job/RoundRobinActivateJobsHandler.java]: This was the issue 👍

**Comment by berkaycanbc** [comment]
@korthout, there is a dependency failure in build. But we do not update any dependencies. 🤔 

Did you face with a similar issue before?

**Comment by korthout** [comment]
> @korthout, there is a dependency failure in build. But we do not update any dependencies. 🤔
> 
> Did you face with a similar issue before?

@berkaycanbc Yes, you're importing a class from a dependency that is not declared in that module: `import org.springframework.util.unit.DataSize;`

```
[INFO] --- dependency:3.3.0:analyze-only (analyze-dependencies) @ zeebe-gateway ---
[WARNING] Used undeclared dependencies found:
[WARNING]    org.springframework:spring-core:jar:5.3.31:compile
[INFO] Add the following to your pom to correct the missing dependencies:
[INFO]
<dependency>
  <groupId>org.springframework</groupId>
  <artifactId>spring-core</artifactId>
  <version>5.3.31</version>
</dependency>
```

So, you'll need to add the following dependency to the gateway's pom file:
```xml
<dependency>
  <groupId>org.springframework</groupId>
  <artifactId>spring-core</artifactId>
</dependency>
```

If we look at the history of that pom file, we'll find 1170acb71039b941fdf9777490fad51f6ca4ff68 adding it to the gateway which was first available in 8.2.0. That's why it's not on 8.1 yet.

**Comment by berkaycanbc** [comment]
Thanks @korthout! 🚀 It was a very tricky one to find out in the build. Updated. 👍

**Review by korthout — APPROVED** [review]
Thanks @berkaycanbc 🥇 

LGTM 👍

**Pull request merged**